src/main/java/com/volasoftware/tinder/services/implementations/FriendServiceImpl.java

```diff
@@ -0,0 +1,71 @@
+package com.volasoftware.tinder.services.implementations;
+
+import com.volasoftware.tinder.constants.AccountConstant;
+import com.volasoftware.tinder.exceptions.AccountNotFoundException;
+import com.volasoftware.tinder.exceptions.FriendExistException;
+import com.volasoftware.tinder.exceptions.FriendNotFoundException;
+import com.volasoftware.tinder.models.Account;
+import com.volasoftware.tinder.repositories.AccountRepository;
+import com.volasoftware.tinder.services.contracts.FriendService;
+import java.util.HashSet;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class FriendServiceImpl implements FriendService {
+
+    private final AccountRepository accountRepository;
+
+    @Override
+    public void addFriend(Long friendId) {
+        Account loggedAccount = getLoggedAccount();
+        Account friendAccount = getAccountById(friendId);
+
+        if (hasFriends(loggedAccount) && friendExist(loggedAccount, friendAccount)) {
+            throw new FriendExistException(AccountConstant.ALREADY_FRIEND);
+        }
+
+        loggedAccount.setFriends(Set.of(friendAccount));
+        accountRepository.save(loggedAccount);
+    }
+
+    @Override
+    public void removeFriend(Long friendId) {
+        Account loggedAccount = getLoggedAccount();
+        Account friendAccount = getAccountById(friendId);
+
+        if (hasFriends(loggedAccount) && friendExist(loggedAccount, friendAccount)) {
+            loggedAccount.getFriends().remove(friendAccount);
+            accountRepository.save(loggedAccount);
+        } else {
+            throw new FriendNotFoundException(AccountConstant.FRIEND_NOT_FOUND);
+        }
+    }
+
+    private boolean hasFriends(Account loggedAccount) {
+        return loggedAccount.getFriends() != null;
+    }
+
+    private boolean friendExist(Account loggedAccount, Account friendAccount) {
+        return loggedAccount.getFriends().contains(friendAccount);
```

Comment:
```suggestion
        return hasFriends && loggedAccount.getFriends().contains(friendAccount);
```
or 
```suggestion
        return !CollectionUtils.isEmpty(loggedAccount.getFriends()) && loggedAccount.getFriends().contains(friendAccount);
```

Reply:
Its not good idea to use hasFriends inside the friendsExists, because this way, you can`t tell wether the collection of friends is null or not.

Reply:
For the addFriend method both check are needed. For the removeFriend method, they can be combined